### PR TITLE
Add support for Slack Markdown in Attachments

### LINKED ISSUE #9: Add support for Markdown in Attachments text/pretext/field fields
RoboSlack messages with Markdown in Attachment pretext/text/fields fields will not render correctly in Slack. We need to implement the `mrkdwn_in` flag: https://api.slack.com/docs/message-formatting#message_formatting .

### PR DESCRIPTION
**Pull request by elektron9**
* Adds support for the `mrkdwn_in` field of a Message `Attachment`. Adds test cases to verify the addition.
* Fixes strikethrough Markdown character.

Fixes #9.

CC @dotCipher for comments.

### CODE REVIEW

File: roboslack-api/src/main/java/com/palantir/roboslack/api/attachments/Attachment.java [hunk 1]
```diff
@@ -201,4 +187,66 @@ public Footer footer() {
...
+            mrkdwnIns.add(MrkdwnIn.PRETEXT.value());
+        }
+
+        // check if the text contains Markdown.
+        if (text().isPresent() && MorePreconditions.containsMarkdown(text().get())) {
+            mrkdwnIns.add(MrkdwnIn.TEXT.value());
+        }
+
+        // check if any of the Fields' values contain Markdown.
+        for (Field field : fields()) {
```
Comment (dotCipher) [inline roboslack-api/src/main/java/com/palantir/roboslack/api/attachments/Attachment.java]: nit: one line, ie:
```
fields().stream().map(Field::value).filter(MorePreconditions::containsMarkdown)
                .findFirst()
                .ifPresent(ignored -> mrkdwnIns.add(MrkdwnIn.FIELDS.value()));
```

File: roboslack-api/src/main/java/com/palantir/roboslack/api/markdown/MrkdwnIn.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    PRETEXT("pretext"),
+    /**
+     * Text.
+     */
+    TEXT("text"),
+    /**
+     * Fields.
+     */
+    FIELDS("fields");
```
Comment (dotCipher) [inline roboslack-api/src/main/java/com/palantir/roboslack/api/markdown/MrkdwnIn.java]: The `String value` property is probably not needed, just need to `toLowerCase()` the `value()`

File: roboslack-api/src/main/java/com/palantir/roboslack/api/attachments/Attachment.java [hunk 3]
```diff
@@ -201,4 +187,65 @@ public Footer footer() {
...
 
+    /**
+     * A special list of flags that tells Slack where to expect Markdown in an Attachment.
+     * Valid values are ["pretext", "text", "fields"].
+     *
+     * @return the {@code mrkdwn_in} list
+     */
+    @Value.Derived
+    @JsonProperty(MRKDWN_IN_FIELD)
+    public Collection<String> markdownInputs() {
```
Comment (dotCipher) [inline roboslack-api/src/main/java/com/palantir/roboslack/api/attachments/Attachment.java]: Actually, why not just use the enum type here? `Collection<MarkdownIn>` ?

**Comment by elektron9** [comment]
@dotCipher fixed, take a look 😄

**Pull request merged**